src/Governance.sol

```diff
@@ -649,10 +650,18 @@ contract Governance is MultiDelegateCall, UserProxyFactory, ReentrancyGuard, Own
 
         // Calculate the offset portions that correspond to each LQTY vote and veto portion
         for (uint256 x; x < _initiatives.length; x++) {
-            absoluteOffsetVotes[x] =
-                _absoluteLQTYVotes[x] * int256(userState.unallocatedOffset) / int256(userState.unallocatedLQTY);
-            absoluteOffsetVetos[x] =
-                _absoluteLQTYVetos[x] * int256(userState.unallocatedOffset) / int256(userState.unallocatedLQTY);
+            // Either _absoluteLQTYVotes[x] or _absoluteLQTYVetos[x] is guaranteed to be zero
+            (int256[] calldata lqtyAmounts, int256[] memory offsets) = _absoluteLQTYVotes[x] > 0
+                ? (_absoluteLQTYVotes, absoluteOffsetVotes)
+                : (_absoluteLQTYVetos, absoluteOffsetVetos);
+
+            uint256 lqtyAmount = uint256(lqtyAmounts[x]);
+            uint256 offset = userState.unallocatedOffset * lqtyAmount / userState.unallocatedLQTY;
+
+            userState.unallocatedLQTY -= lqtyAmount;
+            userState.unallocatedOffset -= offset;
```

Comment:
I guess it’s fine (specially because the difference should be almost negligible), but I wonder if we should put a comment mentioning that this is “path dependent” (related to the order of the array).
Let’s put an example with numbers. Let’s say unallocated LQTY is 10, unallocated offset is 5, and we allocate 3 and 7.
Before, we would have:
- offset[0] = floor(3*5/10) = floor(1.5) = 1
- offset[1] = floor(7*5/10) = floor(3.5) = 3

So there would be a remainder of 1 for the offset, when it should be zero.

Now, we’ll have:
- offset[0] = floor(3*5/10) = floor(1.5) = 1
- offset[1] = floor(7*4/7) = floor(4) = 4 
(as we first reduce unallocated amounts, from 10/5 to 7/4)

So now the final offset is zero as we want, but all the error is “accumulated” in the last initiative.

Reply:
Yes, it is path-dependent. I agree it makes sense to document this.

I wouldn't say that all the error is accumulated in the last initiative, more like spread out over all initiatives. Consider an example with more than 2 initiatives. Let's say we have 4 LQTY and an offset of 3, which we want to split up equally between 4 initiatives.

Ideal:
```solidity
offset[0] = 3*1/4 = 0.75
offset[1] = 3*1/4 = 0.75
offset[2] = 3*1/4 = 0.75
offset[3] = 3*1/4 = 0.75
```

Before:
```solidity
offset[0] = floor(3*1/4) = floor(0.75) = 0 // error[0] = 0 - 0.75 = -0.75
offset[1] = floor(3*1/4) = floor(0.75) = 0 // error[1] = 0 - 0.75 = -0.75
offset[2] = floor(3*1/4) = floor(0.75) = 0 // error[2] = 0 - 0.75 = -0.75
offset[3] = floor(3*1/4) = floor(0.75) = 0 // error[3] = 0 - 0.75 = -0.75
```

Now:
```solidity
offset[0] = floor(3*1/4) = floor(0.75) = 0 // error[0] = 0 - 0.75 = -0.75
offset[1] = floor(3*1/3) = floor(1.00) = 1 // error[1] = 1 - 0.75 =  0.25
offset[2] = floor(2*1/2) = floor(1.00) = 1 // error[2] = 1 - 0.75 =  0.25
offset[3] = floor(1*1/1) = floor(1.00) = 1 // error[3] = 1 - 0.75 =  0.25
```

Reply:
Yes, right! Which means the issue is even less bad.

Reply:
I am now wondering if this is an improvement at all. Sure, the "average" offset error when allocating all of one's LQTY is zero, but it doesn't change the fact that there are (small) errors in the offsets allocated to the individual initiatives. Previously, the per-initiative error was in the `(-1, 0]` range, whereas now it's somewhere in `(-1, 1)`.

Is this worth it? It's not like you can continue accumulating "dust" in `unallocatedOffset`, as we force you to fully deallocate before making any new allocations, which "resets" the accumulated error.

Reply:
Yes, good point, but I still think that having `unallocatedLQTY` zero, but non-zero `unallocatedOffset` feels a little bit ugly. I wonder if having a positive error in the offset of the initiatives can introduce any new problem.